client/landing/gutenboarding/stores/mapped-types.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Mapped types
+ *
+ * This module should only contain mapped types, operations useful in the type system
+ * that do not produce any runtime code.
+ *
+ * Mapped types can be thought of as functions in the type system, they accept some type
+ * argument and transform it to another type.
```

Comment:
Total nitpick: That's kind of the reason why I called the file `type-helpers.ts` rather than `mapped-types.ts` :grimacing: I thought a file named `functions.js` would normally be deemed a bit too non-descript.

(I guess it makes sense in the types ecosystem, where this is just some other level of abstraction that we might not want to conflate with other type helpers/utils :thinking: )

Reply:
We already have `types` modules, which produce some runtime code, for example enums as JS objects. Thinking about sharing this code, I wanted to make sure we're very clear about not producing runtime code and a mapped types restriction seemed like a convenient way to do that.